media/transcoder/src/test/java/com/example/transcoder/CreateJobFromPresetTest.java

```diff
@@ -0,0 +1,153 @@
+import java.nio.file.Paths;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class CreateJobFromPresetTest {
+
+  private static final String LOCATION = "us-central1";
+  private static final String BUCKET_NAME = "java-samples-transcoder-test";
+  private static final String TEST_FILE_NAME = "ChromeCast.mp4";
+  private static final String TEST_FILE_PATH =
+      "src/test/java/com/example/transcoder/testdata/" + TEST_FILE_NAME;
+  private static final String INPUT_URI = "gs://" + BUCKET_NAME + "/" + TEST_FILE_NAME;
+  private static final String OUTPUT_URI_FOR_PRESET =
+      "gs://" + BUCKET_NAME + "/test-output-preset/";
+  private static final String PRESET = "preset/web-hd";
+  private static String PROJECT_ID;
+  private static String PROJECT_NUMBER;
+  private static String JOB_ID;
+  private final PrintStream originalOut = System.out;
+  private ByteArrayOutputStream bout;
+
+  private static String requireEnvVar(String varName) {
+    String varValue = System.getenv(varName);
+    assertNotNull(
+        String.format("Environment variable '%s' is required to perform these tests.", varName));
+    return varValue;
+  }
+
+  private static void deleteBucket(String bucketName) {
+    Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
+    Bucket bucket = storage.get(bucketName);
+    if (bucket != null) {
+      Page<Blob> blobs = bucket.list();
+
+      for (Blob blob : blobs.iterateAll()) {
+        System.out.println(blob.getName());
+        storage.delete(bucketName, blob.getName());
+      }
+      bucket.delete();
+    }
+  }
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    PROJECT_ID = requireEnvVar("GOOGLE_CLOUD_PROJECT");
+    PROJECT_NUMBER = requireEnvVar("GOOGLE_CLOUD_PROJECT_NUMBER");
+  }
+
+  @Before
+  public void beforeTest() throws IOException {
+    bout = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(bout));
+
+    bout = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(bout));
+
+    deleteBucket(BUCKET_NAME);
+    Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
+    storage.create(
+        BucketInfo.newBuilder(BUCKET_NAME)
+            .setStorageClass(StorageClass.STANDARD)
+            .setLocation(LOCATION)
+            .build());
+
+    BlobId blobId = BlobId.of(BUCKET_NAME, TEST_FILE_NAME);
+    BlobInfo blobInfo = BlobInfo.newBuilder(blobId).build();
+    Path path = Paths.get(TEST_FILE_PATH);
+    storage.create(blobInfo, Files.readAllBytes(path));
+    bout.reset();
+  }
+
+  @Test
+  public void test_CreateJobFromPreset() throws Exception {
+    String jobName = String.format("projects/%s/locations/%s/jobs/", PROJECT_NUMBER, LOCATION);
```

Comment:
Why is this using PROJECT_NUMBER?  It's supposed to take both and our samples prefer PROJECT_ID.

Reply:
The API response includes the project number in place of the project id per https://google.aip.dev/cloud/2510.